enacts/onset/agronomy.py

```diff
@@ -157,14 +173,26 @@ def soil_plant_water_balance(
     et_crop = et_crop0.expand_dims({time_dim: peffective[time_dim]}).copy()
     # Filling/emptying bucket day after day
     for doy in range(1, peffective[time_dim].size):
-        if kc_params is not None and planting_date is not None:
-            planted_since = planted_since + pd.Timedelta(days=1)
-            kc = kc_inflex.interp(kc_periods=planted_since, kwargs={"fill_value": 1})
+        if kc_params is not None:
+            if planting_date is not None:
+                planted_since = planted_since + np.timedelta64(1, "D")
+            else:
+                planted_since = planted_since.where(
+                    lambda x: np.isnat(x) == False,
+                    other=xr.where(
+                        sm.isel({time_dim: doy - 1}) >= sm_threshold, -1, np.nan
+                    ).astype("timedelta64[D]"),
+                ) + np.timedelta64(1, "D")
+            kc = kc_inflex.interp(
+                kc_periods=planted_since, kwargs={"fill_value": 1}
+            ).where(lambda x: np.isnan(x) == False, other=1)
         et_crop[{time_dim: doy}] = kc * et.isel({time_dim: doy}, missing_dims='ignore')
         sm[{time_dim: doy}], drainage[{time_dim: doy}] = soil_plant_water_step(
             sm.isel({time_dim: doy - 1}),
             peffective.isel({time_dim: doy}),
             et_crop.isel({time_dim: doy}),
             taw,
         )
-    return sm, drainage, et_crop
\ No newline at end of file
+    if kc_params is not None and planting_date is None:
+        planting_date = peffective[time_dim][doy].drop_vars(time_dim) - planted_since
```

Comment:
Why did you choose not to put this inside the previous if/else, e.g. after line 185? Repeating the if/else test in line 196 makes it less obvious that this only happens in cases where line 185 is reached.

Reply:
Because:

1. I assign planting_date here so I would not be able to use it to check if we are in the case where input planting_date is None, in which case we must go in the part of the algorithms that seeks to update planting_since wherever it can. I could call that something other than planting_date though;
2. I need not the date during each step of the loop so I just do it at the end to provide planting_date as an output. The output planting_date are the ones found by the algo when the input planting_date is None; they are the ones given as input when they were given.

Reply:
Indeed, I missed the fact that the previous test was inside the loop.